test/unit/puppetlabs/jruby_utils/lockable_pool_test.clj

```diff
@@ -584,7 +584,24 @@
     (let [pool (create-populated-pool 1)
           pill (str "i'm a pill")]
       (.insertPill pool pill)
-      (is (identical? (.borrowItem pool) pill)))))
+      (is (identical? (.borrowItem pool) pill))
+      (testing "subsequent borrows return the same pill"
+        (is (identical? (.borrowItem pool) pill)))))
+  (testing "when borrow is blocked, inserting a pill unblocks it"
+    (let [pool (create-populated-pool 1)
+          pill (str "I'm just a pill, yes I'm only a pill")
+          instance (.borrowItem pool)
+          borrow-promise (promise)
+          blocked-borrow (future (let [instance (.borrowItem pool)]
+                                   (deliver borrow-promise instance)))]
+      (is (= 0 (.size pool)))
+      ; Pool is empty and the borrow future is blocked
+      (is (not (realized? borrow-promise)))
```

Comment:
Yeah, the `about-to-borrow-promise` is just a tactic for trying to make the code a bit less racy but definitely not perfect either.  Maybe just a short sleep (like 100 ms) without the promise would help increase the likelihood that we've gotten into the borrow before the pill is inserted without making the tests onerously slower to run.  Whatever you want to do there is fine with me.